octodns/provider/yaml.py

```diff
@@ -102,7 +107,97 @@ def _apply(self, plan):
         if not isdir(self.directory):
             makedirs(self.directory)
 
+        self._do_apply(desired, data)
+
+    def _do_apply(self, desired, data):
         filename = join(self.directory, '{}yaml'.format(desired.name))
         self.log.debug('_apply:   writing filename=%s', filename)
         with open(filename, 'w') as fh:
             safe_dump(dict(data), fh)
+
+
+# Any record name added to this set will be included in the catch-all file,
+# instead of a file matching the record name.
+_CATCHALL_RECORD_NAMES = ('*', '')
```

Comment:
Would prefer this get moved to be a property on `SplitYamlProvider`, and it can lose the `_` and be referred to as `self.CATCHALL_RECORD_NAMES` then.